src/BaseFactory.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry;
+
+use Faker;
+use Zenstruck\Foundry\Exception\FoundryNotBootedException;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ *
+ * @immutable
+ *
+ * @template T
+ *
+ * @method static list<T> createMany(int $number, Attributes $attributes = [])
+ */
+abstract class BaseFactory
+{
+    private static FactoryManager|null $factoryManager = null;
+    private static Configuration|null $configuration = null;
+
+    /** @var array<callable():Parameters> */
+    private array $attributes = [];
+
+    /**
+     * @param Attributes|string $attributes
+     */
+    final public static function new(array|callable|string $attributes = [], string ...$states): static
+    {
+        if (\is_string($attributes)) {
+            $states = [$attributes, ...$states];
+            $attributes = [];
+        }
```

Comment:
I've discovered this thing... never used these string states. This was not in your foundry-next skeleton, would you like that we add a deprecation?

Reply:
done. I've also removed the related doc